tsdb/chunks/head_chunks.go

```diff
@@ -870,6 +870,12 @@ func (cdm *ChunkDiskMapper) Truncate(mint int64) error {
 		cdm.CutNewFile()
 	}
 	errs.Add(cdm.deleteFiles(removedFiles))
+
+	if cdm.curFileSequence == 0 && len(chkFileIndices) == len(removedFiles) {
+		// All files were deleted on startup. Reset the current sequence.
+		cdm.evtlPos.setSeq(0)
```

Comment:
Do we want to do that even if `cdm.deleteFiles()` returned an error?
I guess one could argue that if `deleteFiles()` returns an error then we're in a really bad place anyway, but I think we should still try to keep the `evtlPos` correct.

Reply:
To update `cdm.evtlPos` the lock `cdm.evtlPosMtx` should be held

Reply:
Update the code to take care of deleteFiles failure